internal/operators/ocs/validations.go

```diff
@@ -2,247 +2,251 @@ package ocs
+func (o *operator) validateRequirements(cluster *models.Cluster) (api.ValidationStatus, string) {
 	var status string
-	var err error
-	var inventoryMissing bool
 	hosts := cluster.Hosts
+	numAvailableHosts := int64(len(hosts))
 
-	if int64(len(hosts)) < o.OCSRequiredHosts {
-		status = "Insufficient hosts to deploy OCS. A minimum of 3 hosts is required to deploy OCS. "
-		o.log.Info("Validate Requirements status ", status)
+	if numAvailableHosts < o.config.OCSRequiredHosts {
+		status = "Insufficient hosts to deploy OCS. A minimum of 3 hosts is required to deploy OCS."
+		o.log.Info("OCS requirements validation status ", status)
+		return api.Failure, status
+	}
+	if numAvailableHosts > o.config.OCSRequiredHosts && numAvailableHosts < o.config.OCSMasterWorkerHosts {
+		status = "Insufficient hosts for OCS installation. A cluster with only 3 masters or with a minimum of 3 workers is required"
+		o.log.Info("OCS requirements validation status ", status)
 		return api.Failure, status
 	}
-	var cpuCount int64 = 0       //count the total CPUs on the cluster
-	var totalRAM int64 = 0       // count the total available RAM on the cluster
-	var diskCount int64 = 0      // count the total disks on all the hosts
-	var hostsWithDisks int64 = 0 // to determine total number of hosts with disks. OCS requires atleast 3 hosts with non-bootable disks
-	var insufficientHosts []string
-	if int64(len(hosts)) == o.OCSRequiredHosts { // for only 3 hosts, we need to install OCS in compact mode
-		for _, host := range hosts {
-			inventoryMissing, err = o.nodeResources(host, &cpuCount, &totalRAM, &diskCount, &hostsWithDisks, &insufficientHosts)
-			if err != nil {
-				o.log.Fatal("Error occured while calculating Node requirements ", err)
-				return api.Failure, "Error occured while calculating Node requirements "
-			} else if inventoryMissing {
-				return api.Pending, "Missing Inventory in some of the hosts"
-			}
-		}
 
-	} else if len(hosts) < o.OCSMasterWorkerHosts { // not supporting OCS installation for 2 Workers and 3 Masters
-		status = "Not supporting OCS Installation for 3 Masters and 2 Workers"
-		o.log.Info("Validate Requirements status ", status)
+	nodesResourceInfo := &resourceInfo{}
+	status, err := o.computeNodeResource(cluster, nodesResourceInfo)
+	if err != nil {
+		if nodesResourceInfo.missingInventory {
+			return api.Pending, status
+		}
 		return api.Failure, status
-	} else {
-		for _, host := range hosts { // if the worker nodes >=3 , install OCS on all the worker nodes if they satisfy OCS requirements
+	}
+	canDeployOCS, status := o.validateAggregateRequirements(hosts, nodesResourceInfo)
+
+	o.log.Info(status)
+
+	if canDeployOCS {
+		// this will be used to set count of StorageDevices in StorageCluster manifest
+		o.config.OCSDisksAvailable = nodesResourceInfo.numDisks
+		return api.Success, status
+	}
+	return api.Failure, status
+}
+
+func (o *operator) computeNodeResource(cluster *models.Cluster, nodeInfo *resourceInfo) (string, error) {
```

Comment:
Please rename this function to indicate that it is working on multiple nodes.
Perhaps checkResourcesAllNodes or computeResourcesAllNodes